src/hooks/use-starknet-wallet.js

```diff
@@ -0,0 +1,111 @@
+import {
+  ChainInfo,
+  ChainType,
+  ChainTypeL2,
+  WalletErrorType,
+  WalletStatus
+} from '@starkware-industries/commons-js-enums';
+import {
+  connect as getStarknetWallet,
+  disconnect as resetStarknetWallet,
+  getStarknet
+} from '@starkware-industries/commons-js-libs/get-starknet';
+import {useState} from 'react';
+import {getLogger} from '../utils/logger';
+
+export const useStarknetWallet = () => {
+  const AUTO_CONNECT = process.env.REACT_APP_AUTO_CONNECT;
+  const [error, setError] = useState(null);
```

Comment:
I'm not sure that all of these states are needed. I added a `StarknetWalletButton` in the faucet project without all of it
check `src/providers/StarknetWalletProvider/StarknetWalletProvider.js` and tell me what you think

Reply:
that's because you didn't add `useStarknetWallet` hook, but the `StarknetWalletProvider` works directly with `getStarknetWallet` from 'libs'. I've added `useStarknetWallet` like we have in the webapp